lib/Gateway.js

```diff
@@ -1031,57 +1130,68 @@ Gateway.prototype.discoverDevice = function (node, hassDevice) {
       'Error while discovering device %s of node %d: %s',
       hassID,
       node.id,
-      error.message
+      error
     )
   }
 }
 
-Gateway.prototype.discoverValue = function (node, valueId) {
-  if (this.discovered[valueId.id] || valueId.genre !== 'user') return
+/**
+ * Try to guess the best way to discover this valueId in Hass
+ *
+ * @param {NodeObj} node Internal node object
+ * @param {String} vId value id without the node prefix
+ */
+Gateway.prototype.discoverValue = function (node, vId) {
+  const valueId = node.values[vId]
+
+  if (!valueId || this.discovered[valueId.id]) return
 
   try {
     const result = this.valueTopic(node, valueId, true)
 
-    const topic = result.topic
-
-    if (!topic) return
+    if (!result.topic) return
 
     const valueConf = result.valueConf
 
+    const getTopic = this.mqtt.getTopic(result.topic)
+    const setTopic = result.targetTopic
+      ? this.mqtt.getTopic(result.targetTopic, true)
+      : null
+
     const nodeName = getNodeName(node)
 
-    const type = Constants.commandClass(valueId.commandClass)
     let cfg
 
-    switch (type) {
-      case 'switch_binary':
-      case 'switch_all':
-      case 'switch_toggle_binary':
-        if (valueId.index === 0) {
-          const rgb = node.values['51-1-0']
-          if (rgb) {
-            cfg = copy(hassCfg.light_rgb_switch)
-            cfg.discovery_payload.rgb_state_topic = this.mqtt.getTopic(
-              this.valueTopic(node, rgb)
-            )
-            cfg.discovery_payload.rgb_command_topic =
-              cfg.discovery_payload.rgb_state_topic + '/set'
-          } else {
-            cfg = copy(hassCfg.switch)
-          }
+    const cmdClass = valueId.commandClass
+
+    switch (cmdClass) {
+      case CommandClasses['Binary Switch']:
+      case CommandClasses['All Switch']:
+      case CommandClasses['Binary Toggle Switch']:
+        if (valueId.isCurrentValue) {
+          // const rgb = node.values['51-1-0']
```

Comment:
is this a `TODO`? or can we remove the code?

Reply:
it's a TODO :)

Reply:
https://github.com/zwave-js/node-zwave-js/issues/806

Reply:
added a todo